opal/include/opal_config_bottom.h

```diff
@@ -568,6 +569,17 @@ typedef struct {
 } opal_short_float_complex_t;
 #    endif
 
+/* gcc 13 does not define SSIZE_MAX as required by the POSIX standard.
+ * As a workaround we define ours.
```

Comment:
Just to calrify: is this a gcc 13 issue, a gcc 13.2 issue, a macports 13.2.0 issue, or ...?

Also, is `SSIZE_MAX` guaranteed to be a macro?  E.g., could it be an actual symbol such that `#ifndef SSIZE_MAX` would be false, but using `SSIZE_MAX` would still be fine?

Reply:
According to POSIX, SSIZE_MAX is supposed to be a constant expression suitable for use in #if. In practice we found an instance where it does not exists, so I imagine it could be possible that it might not be a constant expression.

Reply:
This one is a bit tricky...
It works on my Mac with Command Line Tools 13.2 and GCC 13.2.0 provided by brew
But it does not work on an other Mac with Command Line Tools 14.0 and GCC 13.2.0 provided by brew

At first glance, this is because Command Line Tools 14.0 does not provide `limits.h` any more.
That being said, from `/usr/local/Cellar/gcc/13.2.0/lib/gcc/current/gcc/x86_64-apple-darwin22/13/include/limits.h`
```
#ifdef _GCC_NEXT_LIMITS_H
#include_next <limits.h>                /* recurse down to the real one */
#endif
```

I would expect `_GCC_NEXT_LIMITS_H` is **not** defined when recent Command Line Tools are used.
But if I manually `#undef _GCC_NEXT_LIMITS_H`, the `SSIZE_MAX` macro is still undefined.

I've heard several folks from `brew` complaining Apple basically does not care about these issues, so we might want to be pragmatic here and merge this PR.